hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/util/EvaluatorUtils.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.util;
+
+import org.apache.hudi.common.util.ValidationUtils;
+import org.apache.hudi.source.evaluator.AlwaysFalse;
+import org.apache.hudi.source.evaluator.And;
+import org.apache.hudi.source.evaluator.EqualTo;
+import org.apache.hudi.source.evaluator.Evaluator;
+import org.apache.hudi.source.evaluator.GreaterThan;
+import org.apache.hudi.source.evaluator.GreaterThanOrEqual;
+import org.apache.hudi.source.evaluator.In;
+import org.apache.hudi.source.evaluator.IsNotNull;
+import org.apache.hudi.source.evaluator.IsNull;
+import org.apache.hudi.source.evaluator.LessThan;
+import org.apache.hudi.source.evaluator.LessThanOrEqual;
+import org.apache.hudi.source.evaluator.Not;
+import org.apache.hudi.source.evaluator.NotEqualTo;
+import org.apache.hudi.source.evaluator.NullFalseEvaluator;
+import org.apache.hudi.source.evaluator.Or;
+
+import org.apache.flink.table.expressions.CallExpression;
+import org.apache.flink.table.expressions.Expression;
+import org.apache.flink.table.expressions.FieldReferenceExpression;
+import org.apache.flink.table.expressions.ResolvedExpression;
+import org.apache.flink.table.expressions.ValueLiteralExpression;
+import org.apache.flink.table.functions.BuiltInFunctionDefinitions;
+import org.apache.flink.table.functions.FunctionDefinition;
+import org.apache.flink.table.types.logical.LogicalType;
+
+import javax.validation.constraints.NotNull;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Utilities for evaluators.
+ */
+public class EvaluatorUtils {
+
+  /**
```

Comment:
Can we rename the clazz to `ExpressionEvaludators` instead and move all the evaludator classes into it as inner classes,

Reply:
I think it's better not move all the evaluator classes into it as inner classes. Actually, I deliberately separate the original inner evaluator classes as the top-level class in this pr because:
1. I would extends each evaluators to support evaluate based on column values. It's not easy to maintain when each evaluator become more longer.
2. After separate utility class and evaluator classes, the code is more clean.

Reply:
Understood, my point is the Evaluator class is not a user API, it is a developer API and I want to hide the complexity of the impls from different evaluators, we only expose the tool class `ExpressionEvaludators` to the invoker, that is the way I can think of the code can be cleaner.

Reply:
Thanks for your explain.
I reserve my opinion on this point. It is also important to maintain maintainability even for the internal implement.
But if you insist on your point, I'm OK to update the pr on your opinion.